lib/operations/collection/get.js

```diff
@@ -1,39 +1,79 @@
 'use strict';
 
-function createList(map, context) {
-	const keys = Object.keys(map);
-	const query = context.query;
-	const result = [];
-
-	for (let i = 0; i < keys.length; i++) {
-		const id = keys[i];
-		const resource = map[id];
+const PAGE_AFTER = 'pageAfter';
+const PAGE_NUM = 'pageNum';
+const PAGE_SIZE = 'pageSize';
 
-		if (context.mayRead(resource) && (!query || (resource.matches && resource.matches(query)))) {
-			result.push(resource);
+function takeProperty(obj, prop) {
+	let retval;
+	if (obj && obj.hasOwnProperty(prop)) {
+		retval = obj[prop];
+		delete obj[prop];
+		// If the property is a numeber stored as a string we convert it to an actual number
+		if (!isNaN(retval)) {
```

Comment:
isNaN can be a bit confusing, especially needing to use it as a double negative like this can be quite confusing. It'll think up an alternative and replace it.